functions/consumer/websocket-consumer/src/main/java/org/springframework/cloud/fn/consumer/websocket/WebsocketConsumerServer.java

```diff
@@ -33,34 +33,36 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
-import org.springframework.beans.factory.annotation.Autowired;
-
 /**
  * Bootstraps a Netty server using the {@link WebsocketConsumerServerInitializer}. Also adds
  * a {@link LoggingHandler} and uses the <code>logLevel</code>
  * from {@link WebsocketConsumerProperties#logLevel}.
  *
  * @author Oliver Moser
  * @author Gary Russell
+ * @author Chris Bono
  */
 public class WebsocketConsumerServer {
 
 	private static final Log logger = LogFactory.getLog(WebsocketConsumerServer.class);
 
 	static final List<Channel> channels = Collections.synchronizedList(new ArrayList<Channel>());
 
-	@Autowired
```

Comment:
This was working before via indirect field injection as part of ...

```java
@Bean
public WebsocketConsumerServer server() {
	return new WebsocketConsumerServer();
}
```
This style of field injection is not considered a best practice and is harder to follow how the fields get populated (unless you know about it). Constructor injection is preferred in this case.

Reply:
That's correct. It might be OK for end-user project. And I would say for this type of configuration I'd prefer a `@Component` and their scanning instead of `@Bean`.
But when we develop a library it is better to stay away from autowiring at all.